core/region.go

```diff
@@ -474,3 +476,54 @@ func (r *region) findEnabledASGByName(name string) *autoScalingGroup {
 	}
 	return nil
 }
+
+func (r *region) sqsSendMessageSpotInstanceLaunch(asgName *string, instanceID *string, instanceState *string) error {
+	inputJSON := "{\"version\":\"0\",\"id\":\"890abcde-f123-4567-890a-bcdef1234567\"," +
+		"\"detail-type\":\"EC2 Instance State-change Notification\",\"source\":\"aws.events\"," +
+		"\"account\":\"\",\"time\":\"" + time.Now().Format(time.RFC3339) + "\"," +
+		"\"region\":\"" + r.name + "\"," +
+		"\"resources\":[\"arn:aws:events:us-east-1:123456789012:rule/SampleRule\"]," +
+		"\"detail\":" +
+		"{\"instance-id\":\"" + *instanceID + "\",\"state\":\"" + *instanceState + "\"}" + "}"
+
+	svc := r.services.sqs
+
+	_, err := svc.SendMessage(
+		&sqs.SendMessageInput{
+			MessageBody:    &inputJSON,
+			MessageGroupId: asgName,
+			QueueUrl:       &r.conf.SQSQueueURL,
+		})
+
+	if err != nil {
+		logger.Printf("%s Error sending spot instance %s launch event message "+
+			"to the SQS Queue %s: %s", r.name, *instanceID, r.conf.SQSQueueURL, err)
+		return err
+	}
+
+	logger.Printf("%s Successfully sent spot instance %s launch event message"+
+		"to the SQS Queue %s", r.name, *instanceID, r.conf.SQSQueueURL)
+
+	return nil
+}
+
+func (r *region) sqsDeleteMessage(instanceID *string) error {
+	svc := r.services.sqs
+
+	_, err := svc.DeleteMessage(
+		&sqs.DeleteMessageInput{
+			QueueUrl:      &r.conf.SQSQueueURL,
+			ReceiptHandle: &r.conf.sqsReceiptHandle,
```

Comment:
is this the receipt handle we set in the ASG code? to me this looks like a different variable, it might not be the same value.

Reply:
yes is the same, and is working correctly.
Sqs send and receive is handled all in the same region of the main Lambda.